android_configure.py

```diff
@@ -0,0 +1,76 @@
+from platform import platform
+import sys
+import os
+
+# TODO: In next version, it will be a JSON file listing all the patches, and then it will iterate through to apply them.
+def patch_android():
+    print("- Patches List -")
+    print("[1] [deps/v8/src/trap-handler/trap-handler.h] related to https://github.com/nodejs/node/issues/36287")
+    if platform().startswith('Linux'):
+        os.system('patch -f ./deps/v8/src/trap-handler/trap-handler.h < ./android-patches/trap-handler.h.patch')
+    print("\033[92mInfo: \033[0m" + "Tried to patch.")
+
+if platform().startswith('Windows'):
+    print('android-configure is not supported on Windows yet.')
+    sys.exit(1)
+
+if len(sys.argv) == 2 and sys.argv[1] == "patch":
+    patch_android()
+    sys.exit(0)
+
+if len(sys.argv) != 4:
+    print("Usage: ./android-configure [patch] <path to the Android NDK> <Android SDK version> <target architecture>")
+    sys.exit(1)
+
+if not os.path.exists(sys.argv[1]) or not os.listdir(sys.argv[1]):
+    print("\033[91mError: \033[0m" + "Invalid path to the Android NDK")
+    sys.exit(1)
+
+if int(sys.argv[2]) < 24:
+    print("\033[91mError: \033[0m" + "Android SDK version must be at least 24 (Android 7.0)")
+    sys.exit(1)
+
+android_ndk_path = sys.argv[1]
+android_sdk_version = sys.argv[2]
+arch = sys.argv[3]
+
+if sys.argv[3] == "arm":
+    DEST_CPU = "arm"
+    TOOLCHAIN_PREFIX = "armv7a-linux-androideabi"
+elif sys.argv[3] == "arm64" or sys.argv[3] == "aarch64":
+    DEST_CPU = "arm64"
+    TOOLCHAIN_PREFIX = "aarch64-linux-android"
+    arch = "arm64"
+elif sys.argv[3] == "x86":
+    DEST_CPU = "ia32"
+    TOOLCHAIN_PREFIX = "i686-linux-android"
+elif sys.argv[3] == "x86_64":
+    DEST_CPU = "x64"
+    TOOLCHAIN_PREFIX = "x86_64-linux-android"
+    arch = "x64"
+else:
+    print("\033[91mError: \033[0m" + "Invalid target architecture, must be one of: arm, arm64, x86, x86_64")
+    sys.exit(1)
+
+print("\033[92mInfo: \033[0m" + "Configuring for " + DEST_CPU + "...")
+
+if platform().startswith('Darwin'):
```

Comment:
```suggestion
if platform.system() == "Darwin":
```

```shell
$ python
Python 3.10.4 (main, May  7 2022, 23:27:17) [Clang 12.0.0 (clang-1200.0.32.21)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from platform import platform
>>> platform()
'macOS-10.16-x86_64-i386-64bit'
```
`platform.platform()` is not start with `Darwin` from python 3.8, introduced by [this commit](https://github.com/python/cpython/commit/ea0ca218b0c28b2af2b1f6a5d3383569de7fc2c1):
> Changed in version 3.8: On macOS, the function now uses [mac_ver()](https://docs.python.org/3/library/platform.html#platform.mac_ver), if it returns a non-empty release string, to get the macOS version rather than the darwin version.